tests/HubRestAPIClient.ts

```diff
@@ -0,0 +1,806 @@
+          components: apiSpec.components,
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          responses: apiSpec.paths["/v1/reactionById"].get.responses as any,
+        });
+        const errors = validator.validateResponse(200, response.data);
+        expect(errors, JSON.stringify(errors)).is.undefined;
+      });
+
+      it("can fetch an existing reaction", async function () {
+        const reaction = await client.getReactionById({
+          fid: 2,
+          targetFid: 1795,
+          targetHash: "0x7363f449bfb0e7f01c5a1cc0054768ed5146abc0",
+          reactionType: ReactionType.Like,
+        });
+        expectDefinedNonNull(reaction);
+      });
+
+      it("returns null for nonexistent cast", async function () {
+        const reaction = await client.getReactionById({
+          fid: 1,
+          targetFid: 2,
+          targetHash: "0x0000000000000000000000000000000000000000",
+          reactionType: ReactionType.Like,
+        });
+        expect(reaction).to.be.null;
+      });
+    });
+
+    describe("#listReactionsByFid", function () {
+      it("validates against OpenAPI spec", async function () {
+        const response = await client.apis.reactions.listReactionsByFid({
+          fid: 2,
+          reactionType: ReactionType.Like,
+        });
+        expect(response.data.messages).to.not.be.empty;
+        const validator = new OpenAPIResponseValidator({
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          responses: apiSpec.paths["/v1/reactionsByFid"].get.responses as any,
+          components: apiSpec.components,
+        });
+        const errors = validator.validateResponse(200, response.data);
+        expect(errors, JSON.stringify(errors)).is.undefined;
+      });
+
+      it("can fetch a reaction by FID", async function () {
+        const pageSize = 1;
+        const reactions = client.listReactionsByFid(2, ReactionType.Like, {
+          pageSize,
+        });
+        const reaction = await reactions.next();
+        expectDefinedNonNull(reaction.value);
+      });
+    });
+
+    describe("#listReactionsByCast", function () {
+      it("validates against OpenAPI spec", async function () {
+        const response = await client.apis.reactions.listReactionsByCast({
+          targetFid: 1795,
+          targetHash: "0x7363f449bfb0e7f01c5a1cc0054768ed5146abc0",
+          reactionType: ReactionType.Like,
+        });
+        expect(response.data.messages).to.not.be.empty;
+        const validator = new OpenAPIResponseValidator({
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          responses: apiSpec.paths["/v1/reactionsByCast"].get.responses as any,
+          components: apiSpec.components,
+        });
+        const errors = validator.validateResponse(200, response.data);
+        expect(errors, JSON.stringify(errors)).is.undefined;
+      });
+
+      it("can fetch a reaction by cast", async function () {
```

Comment:
same here, should fetch multiple reactions to validate pagination

Reply:
Done